src/main/java/apoc/text/Strings.java

```diff
@@ -205,4 +210,133 @@ public String slug(@Name("text") String text, @Name(value = "delim", defaultValu
+    public String capitalizeAll(@Name("text") String text) {
+        String[] parts = text.split(" ");
+
+        StringBuilder output = new StringBuilder();
+
+        for (String part : parts) {
+            output.append( StringUtils.capitalize(part) + " " );
+
+        }
+
+        return output.toString().trim();
+    }
+
+    @UserFunction
+    @Description("apoc.text.decapitalize(text) YIELD value - decapitalize the first letter of the word")
+    public String decapitalize(@Name("text") String text) {
+        return StringUtils.uncapitalize(text);
+    }
+
+    @UserFunction
+    @Description("apoc.text.decapitalizeAll(text) YIELD value - decapitalize the first letter of the word")
+    public String decapitalizeAll(@Name("text") String text) {
+        String[] parts = text.split(" ");
+
+        StringBuilder output = new StringBuilder();
+
+        for (String part : parts) {
+            output.append( StringUtils.uncapitalize(part) + " " );
+
+        }
+
+        return output.toString().trim();
+    }
+
+    @UserFunction
+    @Description("apoc.text.swapCase(text) YIELD value - Swap the case of a string")
+    public String swapCase(@Name("text") String text) {
+        return StringUtils.swapCase(text);
+    }
+
+    @UserFunction
+    @Description("apoc.text.camelCase(text) YIELD value - Convert a string to camelCase")
+    public String camelCase(@Name("text") String text) {
+        text = text.replaceAll("([^A-Za-z0-9])", " ");
+
+        String[] parts = text.split("(\\s)");
+        StringBuilder output = new StringBuilder();
+
+        for (String part : parts) {
+            part = part.toLowerCase().replaceAll("([^a-z0-9]+)", "");
+            output.append( StringUtils.capitalize( part ) );
+        }
+
+        return output.substring(0, 1).toLowerCase() + output.substring(1);
+    }
+
+    @UserFunction
+    @Description("apoc.text.snakeCase(text) YIELD value - Convert a string to snake-case")
+    public String snakeCase(@Name("text") String text) {
+        // Convert Snake Case
+        if ( text.matches("^([A-Z0-9_]+)$") ) {
+            text = text.toLowerCase().replace("_", " ");
+        }
+
+        String[] parts = text.split("(?=[^a-z0-9])");
+        StringBuilder output = new StringBuilder();
+
+        for (String part : parts) {
+            part = part.trim();
+
+            if (part.length() > 0) {
+                if (output.length() > 0) {
+                    output.append("-");
+                }
+
+                output.append( part.toLowerCase().trim().replace("(^[a-z0-9]+)", "-") );
+            }
+        }
+
+        return output.toString().toLowerCase().replaceAll("--", "-");
+    }
+
+    @UserFunction
+    @Description("apoc.text.toUpperCase(text) YIELD value - Convert a string to UPPER_CASE")
+    public String toUpperCase(@Name("text") String text) {
+        String[] parts = text.split("(?=[^a-z0-9])");
```

Comment:
probably use a + here too. How do you restore the chars that are replaced? like `A` ?